java/org/apache/tomcat/util/net/jsse/PEMFile.java

```diff
@@ -56,6 +56,8 @@
 import org.apache.tomcat.util.codec.binary.Base64;
 import org.apache.tomcat.util.file.ConfigFileLoader;
 import org.apache.tomcat.util.res.StringManager;
+import org.ietf.jgss.GSSException;
```

Comment:
Where is this package sourced? It appears to be present in the JRE since at least 1.7. Interpreting the OID bytes in dotted-notation isn't very difficult. If it avoids adding a dependency or relying on a non-mandatory JRE package, I'd prefer a private implementation. If all JREs are required to provide this package, than it's safe to use IMO.

Reply:
It has been present for the past 15 years. I am using that class in my code for 10+ years. They are default. Azul, Temurin, OpenJDK, Oracle, HPE JVMs have it. The SPNEGO authenticator is using it as well: https://github.com/apache/tomcat/blob/9b3c28da7ec8ddbd721b127f7f3b0a7d475ab797/java/org/apache/catalina/authenticator/SpnegoAuthenticator.java#L46

I consider it safe to use.

Reply:
I agree it is safe enough.